int128/groovy-ssh
# Fix user authentication error due to UserInfo

This adds the feature | fixes the bug...

Currently #198 uses UserInfo to append a host key.
UserInfo is used in many classes such as user authentication and it has a side effect.
This does not use UserInfo and instead catches UnknownHostKey of JSchException.

### Steps to verify the fix
1. Set `knownHosts = addHostKey(...)` and run ssh.
2. Password authentication returns `Auth cancel` but it should return `USERAUTH fail` as backward version.


### Backward compatibility

No break change.
